core/src/main/java/objective/taskboard/jira/properties/EpicGuard.java

```
@@ -0,0 +1,74 @@
+package objective.taskboard.jira.properties;
+
+import java.util.Optional;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import objective.taskboard.jira.client.JiraFieldDataDto;
+import objective.taskboard.jira.properties.JiraProperties.CustomField.CustomFieldDetails;
+import objective.taskboard.jira.properties.JiraProperties.IssueType.IssueTypeDetails;
+
+@Component
+class EpicGuard {
+    private static final Logger log = LoggerFactory.getLogger(EpicGuard.class);
+
+    private final JiraProperties jiraProperties;
+    private final IssueTypeValidator issueTypeValidator;
+    private final CustomFieldValidator customFieldValidator;
+
+    @Autowired
+    public EpicGuard(
+              JiraProperties jiraProperties
+            , IssueTypeValidator issueTypeValidator
+            , CustomFieldValidator customFieldValidator
+    ) {
+        this.jiraProperties = jiraProperties;
+        this.issueTypeValidator = issueTypeValidator;
+        this.customFieldValidator = customFieldValidator;
+    }
+
+    @PostConstruct
+    public void verify() {
+        String propertyName = "Epic type id";
+
+        Optional<IssueTypeDetails> optEpicType = jiraProperties.getIssuetype().getEpic();
+        if (!optEpicType.isPresent()) {
+            log.warn(propertyName + " is not configured. Are you sure you didn't miss it?");
+            return;
```

In  this case, if any other properties exist, should throw an error.